spark-connector/spark-connector-runtime/build.gradle.kts

```diff
@@ -10,15 +10,19 @@ plugins {
   alias(libs.plugins.shadow)
 }
 
-dependencies {
-  implementation(project(":spark-connector:spark-connector"))
-}
-
 val scalaVersion: String = project.properties["scalaVersion"] as? String ?: extra["defaultScalaVersion"].toString()
 val sparkVersion: String = libs.versions.spark.get()
 val sparkMajorVersion: String = sparkVersion.substringBeforeLast(".")
+val icebergVersion: String = libs.versions.iceberg.get()
 val baseName = "${rootProject.name}-spark-connector-runtime-${sparkMajorVersion}_$scalaVersion"
 
+dependencies {
+  implementation(project(":clients:client-java-runtime", configuration = "shadow"))
+  implementation(project(":spark-connector:spark-connector"))
+
+  implementation("org.apache.iceberg:iceberg-spark-runtime-${sparkMajorVersion}_$scalaVersion:$icebergVersion")
```

Comment:
Why do we need to package the iceberg runtime jar into spark connector?

Reply:
we'd better not package Iceberg runtime jar, but there are some limits now, we now register Iceberg extensions on startup.  I prefer to do it in #3396

Reply:
Is it a big work? I think it should be fixed in this version, we cannot ship a Spark connector with iceberg runtime packaged in, it will introduce lots of issues.

Reply:
not a big work, but we have to keep consistent about this,
1. if not registering  Iceberg extensions, Iceberg catalogs couldn't be dymatic loaded in the future
2. when not registering Iceberg extensions, I prefer to add a configuration.

Reply:
Please fix this before 0.5.1 is shipped.

Reply:
ok